hbase-server/src/main/java/org/apache/hadoop/hbase/io/HalfStoreFileReader.java

```diff
@@ -357,7 +357,7 @@ public boolean passesKeyRangeFilter(Scan scan) {
         }
       }
     }
-    return this.firstKey;
+    return Optional.ofNullable(this.firstKey);
```

Comment:
Not sure if this has been answered but I think this method will be called multiple times? And we always need to do a wrapping and a null check?